src/Accounts/Authentication/Factories/AuthenticationFactory.cs

```diff
@@ -184,28 +185,23 @@ private static AzPSAuthenticationFailedException AnalyzeMsalException(Exception
             {
                 if (exception is MsalUiRequiredException msalUiRequiredException)
                 {
-                    //There's no official error message for requiring MFA permission, so have to compare UGLY error message
-                    if (msalUiRequiredException.ErrorCode == "invalid_grant" &&
-                        msalUiRequiredException.Message.Contains("you must use multi-factor authentication to access"))
+                    string errorMessage;
+                    string desensitizedMessage;
+                    if (NeedTenantArmPermission(environment, tenantId, resourceId))
                     {
-                        string errorMessage;
-                        string desensitizedMessage;
-                        if (NeedTenantArmPermission(environment, tenantId, resourceId))
-                        {
-                            errorMessage = $"You must use multi-factor authentication to access tenant {tenantId}, please rerun 'Connect-AzAccount' with additional parameter '-TenantId {tenantId}'.";
-                            desensitizedMessage = "MFA is required to access tenant";
-                        }
-                        else
-                        {
-                            errorMessage = $"You must use multi-factor authentication to access resource {resourceId}, please rerun 'Connect-AzAccount' with additional parameter '-AuthScope {resourceId}'.";
-                            desensitizedMessage = "MFA is required to access resource";
-                        }
-                        return new AzPSAuthenticationFailedException(
-                            errorMessage,
-                            msalUiRequiredException.ErrorCode,
-                            originalException,
-                            desensitizedMessage: desensitizedMessage);
+                        errorMessage = $"You must use multi-factor authentication to access tenant {tenantId}, please rerun 'Connect-AzAccount' with additional parameter '-TenantId {tenantId}'.";
```

Comment:
I rewrote the message so it's more aligned to the sceanrio. Please update the second string as well.
```suggestion
                        errorMessage = $"Failed to authenticate silently. User inteaction is required. This may be due to the access policy of multi-factor authentication to access tenant {tenantId}. Please rerun 'Connect-AzAccount' with additional parameter '-TenantId {tenantId}'.";
```

Reply:
Also we should put the msal exception in debug log